python/pypto/language/parser/ast_parser.py

```
@@ -5038,6 +5038,28 @@ def parse_attribute(self, attr: ast.Attribute) -> ir.Expr:
+        #   ctx.rank   / .nranks → pld.comm_ctx.{rank,nranks}(ctx)
+        #
+        # The lookup is on the LHS *IR type*, not on the textual name, so any
+        # alias (function parameter, ``pld.window(...)`` result Var, threaded-in
+        # tensor argument) routes through the same desugar. We only attempt to
+        # parse the LHS as an IR expression when the leading token is a name in
+        # scope or a chained attribute access; module-level lookups like
+        # ``pl.MemorySpace.Vec`` fall through to the closure-eval path below.
+        if self._looks_like_ir_expr_attribute(attr):
+            try:
+                base = self.parse_expression(attr.value)
+            except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)